### ISSUE TITLE
Deprecate @Provides(type=SET)?

### ISSUE THREAD

**Issue by pyricau** [google/dagger#487]
As far as I can tell, these two things are equivalent:

```
@Provides @IntoSet static String provideOneString() {
    return "ABC";
  }
```

and

```
@Provides(type=SET) static String provideOneString() {
    return "ABC";
  }
```

If I'm right... why do we have two approaches to do the exact same thing? That's confusing, Dagger should pick one.

Seems like `@IntoSet` is the new thing, so `@Provides(type=SET)` should be deprecated.

**Comment by gk5885** [comment]
That is absolutely the thing to do.  We were slacking on it b/c we still have 1 or 2 lingering on Dagger 1 internally, but we can work around that.

**Comment by gk5885** [comment]
Fixed in 9fb6fabdc096273df8bf66be6316336e521a2de0
